src/filters/FilterSystem.ts

```diff
@@ -243,16 +250,22 @@ export class FilterSystem implements System
             return;
         }
 
-        const viewPort = renderer.renderTarget.rootViewPort;
-
         // here we constrain the bounds to the viewport we will render too
         // this should not take into account the x, y offset of the viewport - as this is
         // handled by the viewport on the gpu.
         // need to factor in resolutions also..
+        // const viewPort = renderer.renderTarget.rootViewPort;
         bounds
-            .scale(resolution)
-            .fitBounds(0, viewPort.width, 0, viewPort.height)
-            .ceil()
+            .scale(resolution);
+
+        if (clipToViewport)
+        {
+            const viewPort = renderer.renderTarget.rootViewPort;
+
+            bounds.fitBounds(0, viewPort.width, 0, viewPort.height);
+        }
+
+        bounds.ceil()
```

Comment:
Not sure why do we need to scale it, ceil and then scale back..
Do we need it even if we do not clip bounds into viewport?